src/handlers/fundersHandlers.ts

```diff
@@ -65,20 +77,327 @@ const putFunder = async (req: Request, res: Response): Promise<void> => {
+		await createOrUpdateFunderCollaborativeInvitation(db, req, {
+			funderCollaborativeShortCode: funderShortCode,
+			invitedFunderShortCode,
+			invitationStatus: FunderCollaborativeInvitationStatus.PENDING,
+		});
+	res
+		.status(HTTP_STATUS.SUCCESSFUL.CREATED)
+		.contentType('application/json')
+		.send(funderCollaborativeInvitation);
+};
+
+const getSentFunderCollaborativeInvitations = async (
+	req: Request,
+	res: Response,
+): Promise<void> => {
+	if (!isAuthContext(req)) {
+		throw new FailedMiddlewareError('Unexpected lack of auth context.');
+	}
+	const {
+		params: { funderShortCode },
+	} = req;
+	if (!isShortCode(funderShortCode)) {
+		throw new InputValidationError(
+			'Invalid inviter short code.',
+			isShortCode.errors ?? [],
+		);
+	}
+	const paginationParameters = extractPaginationParameters(req);
+	const { offset, limit } = getLimitValues(paginationParameters);
+	const funderCollaborativeInvitations =
+		await loadFunderCollaborativeInvitiationBundle(
+			db,
+			req,
+			funderShortCode,
+			undefined,
+			undefined,
+			limit,
+			offset,
+		);
+	res
+		.status(HTTP_STATUS.SUCCESSFUL.OK)
+		.contentType('application/json')
+		.send(funderCollaborativeInvitations);
+};
+
+const getRecievedFunderCollaborativeInvitations = async (
+	req: Request,
+	res: Response,
+): Promise<void> => {
+	if (!isAuthContext(req)) {
+		throw new FailedMiddlewareError('Unexpected lack of auth context.');
+	}
+	const {
+		params: { funderShortCode },
+	} = req;
+	if (!isShortCode(funderShortCode)) {
+		throw new InputValidationError(
+			'Invalid funder short code.',
+			isShortCode.errors ?? [],
+		);
+	}
+
+	const paginationParameters = extractPaginationParameters(req);
+	const { offset, limit } = getLimitValues(paginationParameters);
+	const funderCollaborativeInvitations =
+		await loadFunderCollaborativeInvitiationBundle(
+			db,
+			req,
+			undefined,
+			funderShortCode,
+			undefined,
+			limit,
+			offset,
+		);
+	res
+		.status(HTTP_STATUS.SUCCESSFUL.OK)
+		.contentType('application/json')
+		.send(funderCollaborativeInvitations);
+};
+
+const patchFunderCollaborativeInvitation = async (
+	req: Request,
+	res: Response,
+): Promise<void> => {
+	if (!isAuthContext(req)) {
+		throw new FailedMiddlewareError('Unexpected lack of auth context.');
+	}
+	const {
+		params: { funderShortCode, invitedFunderShortCode },
+	} = req;
+
+	if (!isShortCode(funderShortCode)) {
+		throw new InputValidationError(
+			'Invalid funder short code.',
+			isShortCode.errors ?? [],
+		);
+	}
+
+	if (!isShortCode(invitedFunderShortCode)) {
+		throw new InputValidationError(
+			'Invalid invited funder short code.',
+			isShortCode.errors ?? [],
+		);
+	}
+
+	const body = req.body as unknown;
+
+	if (!isFunderCollaborativeInvitationPatch(body)) {
+		throw new InputValidationError(
+			'Invalid request body.',
+			isFunderCollaborativeInvitationPatch.errors ?? [],
+		);
+	}
+
+	const { invitationStatus } = body;
+
+	const funderCollaborativeInvitation =
```

Comment:
We also talked about how this should be a new `updateFunderCollaborativeInvitation` method which will use the `updateFoo` pattern / take in a partial AND (importantly) fail if the entity does not already exist.

We are also changing `createOrUpdate` to just be `create...` and it will only succeed if the entity does not already exist.